internal/server/deputy_notes.go

```diff
@@ -0,0 +1,153 @@
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type DeputyHubNotesInformation interface {
+	GetDeputyDetails(sirius.Context, int) (sirius.DeputyDetails, error)
+	GetDeputyNotes(sirius.Context, int) (sirius.DeputyNoteList, error)
+	AddNote(ctx sirius.Context, title, note string, deputyId, userId int) error
+	GetUserDetails(sirius.Context) (sirius.UserDetails, error)
+}
+
+type deputyHubNotesVars struct {
+	Path          string
+	XSRFToken     string
+	DeputyDetails sirius.DeputyDetails
+	DeputyNotes   sirius.DeputyNoteList
+	Error         string
+	Errors        sirius.ValidationErrors
+	Success		  bool
+	SuccessMessage string
+}
+
+type addNoteVars struct {
+	Path      	string
+	XSRFToken 	string
+	Title      	string
+	Note   		string
+	Success   	bool
+	Error    	sirius.ValidationError
+	Errors    	sirius.ValidationErrors
+	DeputyDetails sirius.DeputyDetails
+}
+
+func renderTemplateForDeputyHubNotes(client DeputyHubNotesInformation, tmpl Template) Handler {
+	return func(perm sirius.PermissionSet, w http.ResponseWriter, r *http.Request) error {
+
+		ctx := getContext(r)
+		routeVars := mux.Vars(r)
+		deputyId, _ := strconv.Atoi(routeVars["id"])
+
+		switch r.Method {
+			case http.MethodGet:
+
+				deputyDetails, err := client.GetDeputyDetails(ctx, deputyId)
+				if err != nil {
+					return err
+				}
+				deputyNotes, err := client.GetDeputyNotes(ctx, deputyId)
+				if err != nil {
+					return err
+				}
+
+				//check if there is a success in url
+				urlTrim := strings.TrimPrefix(r.URL.String(), "/deputy/" + strconv.Itoa(deputyId) + "/notes" )
+				var hasSuccess bool
+				if urlTrim == "?success=true" {
+					hasSuccess = true
+				} else {
+					hasSuccess = false
+				}
+
+				vars := deputyHubNotesVars{
+					Path:          r.URL.Path,
+					XSRFToken:     ctx.XSRFToken,
+					DeputyDetails: deputyDetails,
+					DeputyNotes:   deputyNotes,
+					Success: hasSuccess,
+					SuccessMessage: "Note added",
+				}
+
+				return tmpl.ExecuteTemplate(w, "page", vars)
+
+			case http.MethodPost:
+				var vars addNoteVars
+				var (
+					title   = r.PostFormValue("title")
+					note  	= r.PostFormValue("note")
+				)
+
+				userId, err := client.GetUserDetails(ctx)
+				if err != nil {
+					return err
+				}
+
+				deputyDetails, err := client.GetDeputyDetails(ctx, deputyId)
+				if err != nil {
+					return err
+				}
+
+				err = client.AddNote(ctx, title, note, deputyId, userId.ID)
+
+
+				if verr, ok := err.(sirius.ValidationError); ok {
+
+					verr.Errors = renameValidationErrorMessages(verr.Errors)
+
+					vars = addNoteVars{
+						Path:      r.URL.Path,
+						XSRFToken: ctx.XSRFToken,
+						Title:      title,
+						Note:   note,
+						Errors:    verr.Errors,
+						DeputyDetails: deputyDetails,
+					}
+
+					w.WriteHeader(http.StatusBadRequest)
+					return tmpl.ExecuteTemplate(w, "page", vars)
+				} else if err != nil {
+					return err
+				}
+
+				return RedirectError(fmt.Sprintf("/deputy/%d/notes?success=true", deputyId))
```

Comment:
Is this not the success case? Why is it using `RedirectError`?

Reply:
We talked to Josh @hawx   about this, I think it's the existing pattern they implemented in user management that is used in all our Go projects. I think it was the only way to redirect, it's not a "real" error but one they created.

Reply:
If it is not an error, it shouldn't have error in the name. I'd suggest renaming it to `Redirect`.

Reply:
So yeah as you said, it is the success. Found it confusing but decided to keep the name originally because on line 42 the function returns an error. If you still think it is worth renaming, happy to do so 🤔

Reply:
If you found it confusing and I've asked questions about it, the same thing will happen in the future. We shouldn't be leaving confusing code in our code base.